src/averageAges.js

```diff
@@ -14,12 +14,27 @@
  *
  * @return {number}
  */
+// learn how to use array methods like .filter .map .some .every .find .reduce
+// avoid using loop and forEach
+// replace `if ()` statement with &&, || or ?:
+// without nesting
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const diedInCentury = men.filter(man =>
+    Math.ceil(man.died / 100) === century);
+
+  const menAges = century
+    ? diedInCentury.map(man =>
+      man.died - man.born)
+
+    : men.map(man => man.died - man.born);
+
+  const sumMenAge = menAges.reduce((sum, age) => sum + age, 0);
```

Comment:
Let's create a function-helper, that takes an array of people and return their average age. It will help not to duplicate yourself and make your code cleaner